test/AutoRest.TestServerLowLevel.Tests/LowLevel/Generation/ReadonlyPropertyWritingTests.cs

```diff
@@ -209,6 +222,169 @@ public partial class RoundTripModel: global::Azure.Core.IUtf8JsonSerializable
 }
 }
 }
+"
+            }
+        };
+
+        private static readonly object[] OutputModelCase =
+        {
+            new string[]
+            {
+                @"// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Collections.Generic;
+
+namespace Cadl.TestServer.ReadonlyProperties.Models
+{
+/// <summary> Readonly model. </summary>
+public partial class OutputModel
+{
+/// <summary> Initializes a new instance of OutputModel. </summary>
+internal OutputModel()
+{
+RequiredReadonlyStringList = new List<string>(0).AsReadOnly();
+RequiredReadonlyIntList = new List<int>(0).AsReadOnly();
+OptionalReadonlyStringList = new List<string>(0).AsReadOnly();
+OptionalReadonlyIntList = new List<int>(0).AsReadOnly();
+}
+/// <summary> Initializes a new instance of OutputModel. </summary>
+/// <param name=""requiredReadonlyString""> Required string, illustrating a readonly reference type property. </param>
+/// <param name=""requiredReadonlyInt""> Required int, illustrating a readonly reference type property. </param>
+/// <param name=""optionalReadonlyString""> Optional string, illustrating a readonly reference type property. </param>
+/// <param name=""optionalReadonlyInt""> Optional int, illustrating a readonly reference type property. </param>
+/// <param name=""requiredReadonlyStringList""> Required readonly string collection. </param>
+/// <param name=""requiredReadonlyIntList""> Required readonly int collection. </param>
+/// <param name=""optionalReadonlyStringList""> Optional readonly string collection. </param>
+/// <param name=""optionalReadonlyIntList""> Optional readonly int collection. </param>
+internal OutputModel(string requiredReadonlyString,int requiredReadonlyInt,string optionalReadonlyString,int optionalReadonlyInt,global::System.Collections.Generic.IReadOnlyList<string> requiredReadonlyStringList,global::System.Collections.Generic.IReadOnlyList<int> requiredReadonlyIntList,global::System.Collections.Generic.IReadOnlyList<string> optionalReadonlyStringList,global::System.Collections.Generic.IReadOnlyList<int> optionalReadonlyIntList)
+{
+RequiredReadonlyString = requiredReadonlyString;
+RequiredReadonlyInt = requiredReadonlyInt;
+OptionalReadonlyString = optionalReadonlyString;
+OptionalReadonlyInt = optionalReadonlyInt;
+RequiredReadonlyStringList = requiredReadonlyStringList;
+RequiredReadonlyIntList = requiredReadonlyIntList;
+OptionalReadonlyStringList = optionalReadonlyStringList;
+OptionalReadonlyIntList = optionalReadonlyIntList;
+}
+
+/// <summary> Required string, illustrating a readonly reference type property. </summary>
+public string RequiredReadonlyString{ get; }
```

Comment:
As this property is required but 'string' is nullable, how can we make sure the returned value is not null?

Reply:
There is a check in the public constructor for required properties: https://github.com/annelo-msft/azure-sdk-for-net/blob/21fc723ab4307df02263dea79f212edbbce615cd/sdk/template/Azure.Template/src/Generated/Models/RoundTripModel.cs?rgh-link-date=2022-07-22T18%3A28%3A01Z#L22